src/plugins/codeStyle/index.ts

```diff
@@ -18,13 +36,49 @@ export default class CodeStyle {
         extractFixes(addFixes, event.diagnostics);
     }
 
-    afterFileValidate(file: BscFile) {
-        if (!isBrsFile(file) || this.lintContext.ignores(file)) {
-            return;
+    validateXMLFile(file: XmlFile) {
+        const diagnostics: Omit<BsDiagnostic, "file">[] = [];
+        const apiFields = file.parser.ast?.component?.api?.fields;
+        const { noArrayComponentFieldType, noAssocarrayComponentFieldType } = this.lintContext.severity;
+
+        const validateArrayComponentFieldType = noArrayComponentFieldType !== DiagnosticSeverity.Hint;
+        const validateAssocarrayComponentFieldType = noAssocarrayComponentFieldType !== DiagnosticSeverity.Hint;
+
+        if (apiFields) {
+            for (const field of apiFields) {
```

Comment:
we can simplify this section with something more like this to eliminate the `apiFields` named variable and decrease the indentation by 1 level by eliminating the if statement:

```typescript
for (const field of file.parser?.ast?.component?.api?.fields ?? []) {

Reply:
Done